lib/storage/bucket.js

```diff
@@ -410,6 +411,186 @@ Bucket.prototype.getMetadata = function(callback) {
+  var self = this;
+
+  if (util.is(options, 'function')) {
+    callback = options;
+    options = {};
+  }
+
+  options = options || {};
+  options.private = true;
+
+  async.parallel([setPredefinedAcl, makeFilesPrivate], callback);
+
+  function setPredefinedAcl(done) {
+    self.setPredefinedAcl_(options, done);
+  }
+
+  function makeFilesPrivate(done) {
+    if (!options.includeFiles) {
+      done();
+      return;
+    }
+
+    self.makeAllFilesPublicPrivate_(options, done);
+  }
+};
+
+/**
+ * Make the bucket publicly readable.
+ *
+ * You may also choose to make the contents of the bucket publicly readable by
+ * specifying `includeFiles: true`. This will automatically run
+ * {module:storage/file#makePublic} for every file in the bucket.
+ *
+ * When specifying `includeFiles: true`, use `force: true` to delay execution of
+ * your callback until all files have been processed. By default, the callback
+ * is executed after the first error. Use `force` to queue such errors until all
+ * files have been procssed, after which they will be returned as an array as
+ * the first argument to your callback.
+ *
+ * NOTE: This may cause the process to be long-running and use a high number of
+ * requests. Use with caution.
+ *
+ * @param {object=} options - The configuration object.
+ * @param {boolean} options.includeFiles - Make each file in the bucket publicly
+ *     readable. Default: `false`.
+ * @param {boolean} options.force - Queue errors occurred while making files
+ *     public until all files have been processed.
+ * @param {function} callback - The callback function.
+ *
+ * @example
+ * //-
+ * // Make the bucket publicly readable.
+ * //-
+ * bucket.makePublic(function(err) {});
+ *
+ * //-
+ * // Make the bucket and its contents publicly readable.
+ * //-
+ * var opts = {
+ *   includeFiles: true
+ * };
+ *
+ * bucket.makePublic(opts, function(err, files) {
+ *   // `err`:
+ *   //    The first error to occur, otherwise null.
+ *   //
+ *   // `files`:
+ *   //    Array of files successfully made public in the bucket.
+ * });
+ *
+ * //-
+ * // Make the bucket and its contents publicly readable, using force to
+ * // suppress errors until all files have been processed.
+ * //-
+ * var opts = {
+ *   includeFiles: true,
+ *   force: true
+ * };
+ *
+ * bucket.makePublic(opts, function(errors, files) {
+ *   // `errors`:
+ *   //    Array of errors if any occurred, otherwise null.
+ *   //
+ *   // `files`:
+ *   //    Array of files successfully made public in the bucket.
+ * });
+ */
+Bucket.prototype.makePublic = function(options, callback) {
+  var self = this;
+
+  if (util.is(options, 'function')) {
+    callback = options;
+    options = {};
+  }
+
+  options = options || {};
+  options.public = true;
+
+  async.parallel([addAclPermissions, makeFilesPublic], callback);
+
+  function addAclPermissions(done) {
+    // Allow reading bucket contents while preserving original permissions.
+    self.acl.default.add({
```

Comment:
Does this additionally make the bucket listing public readable? I'd assume we'd have to additionally do 
``` js
self.acl.add({
  entity: 'allUsers',
  role: 'READER'
}, cb);
```